File: Service/CountryService.vb
Imports BO
Imports DAL
Imports Facade

Public Class CountryService
    Implements ICountryService

    Public Function GetCountrys() As GetResponse(Of CountryBO) Implements ICountryService.GetCountrys
        Dim response As New GetResponse(Of CountryBO)
        Dim uow As New UnitOfWork(False)
        Dim dao = uow.GetCountryDAO()
        Dim entities = dao.GetNoTracking()
        For Each _entity In entities
            Dim bo As New CountryBO()
            bo.CountryID = _entity.ID
            bo.Name = _entity.LandNaam
            bo.ISOCode = _entity.LandISOCode

            response.AddValue(bo)
        Next
        Return response
    End Function
    Public Function GetCountryById(id As Integer) As GetResponse(Of CountryBO) Implements ICountryService.GetCountryById
        Dim response As New GetResponse(Of CountryBO)
        Dim uow As New UnitOfWork(False)
        Dim dao = uow.GetCountryDAO()
        Dim _entity = dao.GetById(id)

        Dim bo As New CountryBO()
        bo.CountryID = _entity.ID
        bo.Name = _entity.LandNaam
        bo.ISOCode = _entity.LandISOCode

        response.AddValue(bo)
        Return response
    End Function
    Public Function GetVisibleCountriesForSelect() As GetResponse(Of IdNameBO) Implements ICountryService.GetVisibleCountriesForSelect
        Dim response As New GetResponse(Of IdNameBO)
        Dim uow As New UnitOfWork(False)
        Dim dao = uow.GetCountryDAO()
        Dim entities = dao.GetNoTracking().Where(CountryQuery.GetVisibleQuery(True))
        For Each _entity In entities
            response.AddValue(_entity.GetIdName())
        Next
        Return response
    End Function
End Class

File: ServiceCore/CountryService.cs
using FacadeCore;
using BOCore;
using DALCore;
using DALCore.Models;
using DALCore.Query;
using System.Linq;

namespace ServiceCore
{
    public class CountryService : ICountryService
    {
        private readonly UnitOfWorkCore _uow;

        public CountryService(UnitOfWorkCore uow)
        {
            _uow = uow;
        }

        public GetResponse<CountryBO> GetCountrys()
        {
            var response = new GetResponse<CountryBO>();

            var entities = _uow.Countries.GetNoTracking(); // zorg dat UnitOfWorkCore.Countries bestaat
            foreach (var e in entities)
            {
                var bo = new CountryBO
                {
                    CountryId = e.Id,
                    Name = e.LandNaam,
                    ISOCode = e.LandIsocode
                };
                response.AddValue(bo);
            }
            return response;
        }

        public GetResponse<CountryBO> GetCountryById(int id)
        {
            var response = new GetResponse<CountryBO>();

            var e = _uow.Countries.GetById(id);
            if (e == null)
            {
                response.AddError("country not found");
                return response;
            }

            var bo = new CountryBO
            {
                CountryId = e.Id,
                Name = e.LandNaam,
                ISOCode = e.LandIsocode
            };
            response.AddValue(bo);

            return response;
        }

        public GetResponse<IdNameBO> GetVisibleCountriesForSelect()
        {
            var response = new GetResponse<IdNameBO>();

            var entities = _uow.Countries.GetNoTracking()
                .Where(CountryQuery.GetVisibleQuery(true));

            foreach (var e in entities)
                response.AddValue(e.GetIdName());

            return response;
        }
    }
}
